pkg/types/apisix/v1/plugin_types.go

```diff
@@ -177,3 +191,46 @@ func (p *Headers) DeepCopy() *Headers {
 	p.DeepCopyInto(out)
 	return out
 }
+
+// +k8s:deepcopy-gen=true
+type Expr struct {
+	StringVal string
+	ArrayVal  []Expr
+}
+
+func (e *Expr) MarshalJSON() ([]byte, error) {
+	if e.StringVal != "" {
+		return json.Marshal(e.StringVal)
+	} else {
+		return json.Marshal(e.ArrayVal)
+	}
+}
+
+func (expr *Expr) UnmarshalJSON(buf []byte) error {
```

Comment:
It would be better to have a simple test for this function.

Reply:
Hey, I was implementing a simple golang toolkit package called [go-resty-expr](https://github.com/incubator4/go-resty-expr) , 
for generating expressions conforming to the [lua-resty-expr](https://github.com/api7/lua-resty-expr) specification.

Also I've configured the corresponding tests and CI. May I use this package  in this ?

Reply:
> It would be better to have a simple test for this function.

Hey, I was implementing a simple golang toolkit package called [go-resty-expr](https://github.com/incubator4/go-resty-expr) , 
for generating expressions conforming to the [lua-resty-expr](https://github.com/api7/lua-resty-expr) specification.

Also I've configured the corresponding tests and CI. May I use this package  in this ?